src/server/controllers/user.js

```diff
@@ -1,39 +1,69 @@
-const bcrypt = require('bcrypt');
-const jwt = require('jsonwebtoken');
-const { PrismaClient } = require('@prisma/client');
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const { PrismaClient } = require("@prisma/client");
 const prisma = new PrismaClient();
 
-const jwtSecret = 'mysecret';
+const saltRounds = 10;
+const secret = process.env.JWT_SECRET;
 
-const register = async (req, res) => {
-    const { username, password } = req.body;
+const registerUser = async (req, res) => {
+  const { username, password } = req.body;
 
-    const createdUser = null;
+  if (!username || !password) {
+    res.status(400).json({ message: "Please provide a username and password" });
+  }
 
-    res.json({ data: createdUser });
-};
+  bcrypt.genSalt(saltRounds, (err, salt) => {
+    bcrypt.hash(password, salt, async (err, hashedPw) => {
+      try {
+        const newUser = await prisma.user.create({
+          data: {
+            username,
+            password: hashedPw,
+          },
+        });
 
-const login = async (req, res) => {
-    const { username, password } = req.body;
+        delete newUser.password;
 
-    const foundUser = null;
+        res.status(201).json({ status: "success", user: newUser });
+      } catch (error) {
+        if (error.code === "P2002") {
+          res
+            .status(403)
+            .json({ error: `The username ${username} is already taken` });
+        } else {
+          res.status(500).json({ error });
+        }
```

Comment:
error handling is wonderfully done

Reply:
I mean this part is just what Julian showed us, I would love to take credit though ; )